File: Sorting/0324.Wiggle Sort II/Solution.py
class Solution:
    def wiggleSort(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        cnt = [0] * 5010
        for v in nums:
            cnt[v] += 1
        idx, n, j = 0, len(nums), 5000
        for i in range(1, n, 2):
            while cnt[j] == 0:
                j -= 1
            nums[i] = j
            cnt[j] -= 1
        for i in range(0, n, 2):
            while cnt[j] == 0:
                j -= 1
            nums[i] = j
            cnt[j] -= 1

File: Sorting/0324.Wiggle Sort II/Solution.java
class Solution {
    public void wiggleSort(int[] nums) {
        int n = nums.length;
        int[] cnt = new int[5010];
        for (int v : nums) {
            cnt[v]++;
        }
        int j = 5000;
        for (int i = 1; i < n; i += 2) {
            while (cnt[j] == 0) {
                j--;
            }
            nums[i] = j;
            cnt[j]--;
        }
        for (int i = 0; i < n; i += 2) {
            while (cnt[j] == 0) {
                j--;
            }
            nums[i] = j;
            cnt[j]--;
        }
    }
}